Номер появления слова

Maksonchik88 opened the pull request:
*Инициализировал словарь
*Открыл и прочитал файл
*Заполнил словарь слово -> количесво повторений, распечатывая при этом каждое значение в итерации

Maksonchik88 committed:
Номер появления слова

*Инициализировал словарь
*Открыл и прочитал файл
*Заполнил словарь слово -> количесво повторений, распечатывая при этом каждое значение в итерации

iliadmitriev commented:
А ссылку на задачу добавь?

iliadmitriev commented on dict/word_occurrence_number.py:
```diff
@@ -0,0 +1,7 @@
+    for lines in file:
+        lines = lines.split()
+        for line in lines:
+            data[line] = data.get(line, -1) + 1
+            print(data[line], end=' ')
```
для лучшей читаемости кода лучше разделить операцию чтения из словаря и операцию записи в словарь увеличенного на 1 значения.
к стати хорошая идея читать значение из словаря с помощью метода get у словаря.

а что значит константа -1, и почему они именно -1 равна?

Maksonchik88 replied:
Как разделить?

Maksonchik88 replied:
-1 в методе потому, что в условии считать с 0 количество слов в словаре, а не с 1.
data[line] = data.get(line, -1)
            data[line] += 1

Maksonchik88 replied:
и выходит, что если не ключа в словаре, то мы в значении -1 и потом прибавляем к значению 1. Но если ключ есть, то гет возвращает существующее значение и потом мы его переписываем, добавляя единицу к значению

iliadmitriev reviewed (changes requested):
требуется пояснение. что за константа в коде `-1`
P.S. добавь ссылку на задачу в текст PR

Maksonchik88 committed:
https://informatics.msk.ru/mod/statements/view.php?id=4535&chapterid=3759#1

*Описал алгоритм присвоения значения с помощью метода get() в две строки для наглядности
*Коснтанта -1 задает начальное значение по ключу в случае, когда мы инициализируем ключ -> значение в словаре
*Следующим действием мы добавляем к текущему значению единицу и записываем в словрь это значение
*В случае, когда ключ в словаре присутвует, get() возвращает существующее значение, а затем увеличивает счетчик на единицу

Maksonchik88 committed:
Не совсем верно описал логику get()
*когда мы присваиваем значение по ключу впервые, то используем метод гет, так как он без ошибки вернет значение, если нет ключа. А потом мы к этому значению прибаляем елиницу
*когда же мы уже имеем ключ -> значение в словаре, то метод гет возвращает значение уже по ключу, а затем мы прибавляем единицу
